### DISCUSSION TITLE
troubleshots while upgrading to version 7

### DISCUSSION THREAD

**Discussion by hieunm22** [katspaugh/wavesurfer.js#3488] [Q&A]
I'm using this lib to display audio file as replacement built-in html audio tag
Version `6.2.0` was running fine to me. but I had a trouble with test framework, so I had to update to version `^7.2.3`
Here're some issues I have which don't appear in version `6.2.0`
- I can't catch timeupdate event
- I can't set muted to audio element
- My wave component looks weird as below

<img width="379" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/5255023/6ef27964-775e-4754-bd72-d0bb84abfd72">

and here's the wave component looks like in version 6

<img width="416" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/5255023/92b45d4e-eee9-4337-a48d-899a617a56b0">

Thanks

**Comment by katspaugh** [comment]
Please read the v7 upgrade guide in the readme.
If you have any specific questions, I’m happy to answer, especially if you paste your code.

**Comment by hieunm22** [comment]
the muted issue is resolved by using `setMuted` function
and the timeupdate is resolved by updating my inside logic, too.
thanks

The last issue is still remain
- the component with height can't be set by using `setOptions` function which described in the [v7 readme](https://github.com/katspaugh/wavesurfer.js/blob/main/README.md)
```typescript
const wave = WaveSurfer.create({
  container: "#waveform"
})
wave.setOptions({
  backend: "MediaElement",
  barGap: 2,
  barRadius: 3,
  barWidth: 3,
  barHeight: 24,
  cursorColor: "white",
  cursorWidth: 0,
  height: 24,
  progressColor,
  waveColor
})
```
Did I do correct?

**Reply by katspaugh** [reply]
Yes, looks good. Might be a bug if the height isn’t updated.

**Reply by hieunm22** [reply]
thanks. please let me know if the bug is fixed

**Reply by katspaugh** [reply]
I've double-checked, and there's no bug. It sets the height correctly. See this example: https://wavesurfer.xyz/examples/?29e402e400f63236249c634a0805a693

**Reply by hieunm22** [reply]
You changed the html node structure in v7, so it impacted to my wave form UI.
let me double check in my code

**Reply by hieunm22** [reply]
Fixed the issue myself. A css property impacted to my wave from which didn't impact in version 6
Thanks for your support
